src/Nethermind/Nethermind.JsonRpc.Test/Modules/Eth/EthRpcModuleTests.GasPrice.cs

```diff
@@ -0,0 +1,169 @@
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Nethermind.Blockchain.Find;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Specs;
+using Nethermind.Core.Test.Builders;
+using Nethermind.Facade;
+using Nethermind.Int256;
+using Nethermind.JsonRpc.Modules.Eth;
+using Nethermind.Logging;
+using Nethermind.State;
+using Nethermind.TxPool;
+using Nethermind.Wallet;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace Nethermind.JsonRpc.Test.Modules.Eth
+{
+    public partial class EthRpcModuleTests
+    {
+        [Test]
+        public async Task Eth_gasPrice_WhenHeadBlockIsNull_ThrowsException()
+        {
+            using Context ctx = await Context.Create();
+            IBlockFinder blockFinder = Substitute.For<IBlockFinder>();
+            blockFinder.FindHeadBlock().Returns((Block) null);
+            
+            ctx._test = await TestRpcBlockchain.ForTest(SealEngineType.NethDev).WithBlockFinder(blockFinder).Build();
+            string serialized = ctx._test.TestEthRpc("eth_gasPrice");
+            
+            Assert.AreEqual("{\"jsonrpc\":\"2.0\",\"error\":{\"code\":-32603,\"message\":\"Head Block was not found.\"},\"id\":67}",
+                serialized);
+        }
+        
+        [Test]
+        public async Task Eth_gasPrice_GivenValidHeadBlock_CallsGasPriceEstimateFromGasPriceOracle()
+        {
+            Context ctx = await Context.Create(); 
+            IBlockFinder blockFinder = Substitute.For<IBlockFinder>();
+            IGasPriceOracle gasPriceOracle = Substitute.For<IGasPriceOracle>();
+            Block testBlock = Build.A.Block.Genesis.TestObject;
+            blockFinder.FindHeadBlock().Returns(testBlock);
+            blockFinder.FindBlock(Arg.Is<long>(a => a == 0)).Returns(testBlock);
+            
+            ctx._test = await TestRpcBlockchain.ForTest(SealEngineType.NethDev).WithBlockFinder(blockFinder)
+                .WithGasPriceOracle(gasPriceOracle).Build();
+            ctx._test.TestEthRpc("eth_gasPrice");
+            
+            gasPriceOracle.Received(1).GasPriceEstimate(Arg.Any<Block>(), Arg.Any<IBlockFinder>());
+        }
+
+        [Test]
+        public async Task Eth_gasPrice_BlocksAvailableLessThanBlocksToCheck_ShouldGiveCorrectResult()
```

Comment:
Can you add this type of test with 1559 transactions and for all edge cases you mentioned before?